src/Passes/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.4.3)
+
+project(QSharpPasses)
+
+find_package(LLVM REQUIRED CONFIG)
+include(CheckCXXCompilerFlag)
+
+message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
+message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
+
+# Setting the standard configuration for the C++ compiler
+# Rather than allowing C++17, we restrict the ourselves to 
+# C++14 as this is the standard currently used by LLVM. While
+# there is a very small change that the difference in standard 
+# would break things, it is a possibility nonetheless.
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Weverything -Wconversion -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-padded -Wno-exit-time-destructors -Wno-global-constructors")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror ")
+
+# LLVM is normally built without RTTI. Be consistent with that.
+if(NOT LLVM_ENABLE_RTTI)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
+endif()
+
+# -fvisibility-inlines-hidden is set when building LLVM and on Darwin warnings
+# are triggered if llvm-tutor is built without this flag (though otherwise it
+# builds fine). For consistency, add it here too.
+check_cxx_compiler_flag("-fvisibility-inlines-hidden" SUPPORTS_FVISIBILITY_INLINES_HIDDEN_FLAG)
+if (${SUPPORTS_FVISIBILITY_INLINES_HIDDEN_FLAG} EQUAL "1")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility-inlines-hidden")
+endif()
+
+# We export the compile commands which are needed by clang-tidy 
+# to run the static analysis
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
FYI: I assume that with "static analysis" you mean the code formatting tooling (clang-tidy and clang-format), but not the topic discussed [here](https://clang-analyzer.llvm.org/).

Reply:
Clang-tidy is a static analyser more than just a formatter - it also does static analysis and, in fact, it uses the project that you are linking. You can activate different parts of static analyser using flags in the top section of `.clang-tidy`. Here are some details:

https://clang.llvm.org/docs/analyzer/checkers.html

This means that besides checking for correct casing and naming, `clang-tidy` also checks that loops does not use floats `clang-analyzer-security.FloatLoopCounter` for instance.

Reply:
Thank you for explaining!